### DISCUSSION TITLE
TTL for Table records and storage entities

### DISCUSSION THREAD

**Discussion by sandbox-apps** [supabase/supabase#2362] [Questions] [answered]
Common use case of this are facebook stories.

**Comment by dshukertjr** [comment, marked as answer]
Hi @sandbox-apps !

You can achieve this by utilizing row level security of Postgres. 

Here is an example table: 
```sql
create table if not exists public.stories (
    id uuid not null primary key DEFAULT uuid_generate_v4 (),
    created_at timestamp with time zone default timezone('utc' :: text, now()) not null
    -- other columns
);

alter table public.stories enable row level security;
create policy "Stories are live for a day" on public.stories for select using (created_at > (current_timestamp - interval '1 day')); 
```

And here is an example security policy for storage objects:
```sql
insert into storage.buckets (id, name) values ('stories', 'stories');
create policy "Stories are live for a day" on storage.objects for select using (bucket_id = 'stories' and created_at > (current_timestamp - interval '1 day'));
```

With these row level security policies in place, your uses will only be able to access the data for a day! Let me know what you think!

**Reply by fergusmeiklejohn** [reply]
That's a cool example of how to use RLS to deliver a product feature. Examples like this help us to understand what else we might do with RLS and would be useful added to the docs

**Reply by sandbox-apps** [reply]
This can be also useful for deleting row and objects via cron job per minute
`using (created_at > (current_timestamp - interval '1 day')`

**Reply by dshukertjr** [reply]
@fergusmeiklejohn @sandbox-apps 

Thanks for the amazing feedback! This example is now included in the row level policy examples in documentation 🎉
https://supabase.io/docs/guides/auth#time-to-live-for-rows

**Reply by sandbox-apps** [reply]
I never thought just by asking that the setup will be useful for docs. Thanks!! :D
